### PR TITLE
feat(snapshots): delete snapshots

### PR DESCRIPTION
**Pull request by ksendart**
#256

### CODE REVIEW

File: src/app/shared/services/snapshot.service.ts [hunk 1]
```diff
@@ -38,6 +38,10 @@ export class SnapshotService extends BaseBackendCachedService<Snapshot> {
       });
   }
 
+  public markForRemoval(snapshot: Snapshot): Observable<Snapshot> {
+    return this.snapshotTagService.markForRemoval(snapshot) as Observable<Snapshot>;
```
Comment (andrewbents) [inline src/app/shared/services/snapshot.service.ts]: why casting? I think this method return something like success: true. Can you check that?

File: src/app/shared/services/volume.service.ts [hunk 2]
```diff
@@ -132,6 +132,7 @@ export class VolumeService extends BaseBackendService<Volume> {
   }
 
   public markForRemoval(volume: Volume): Observable< Volume> {
+    volume.snapshots.forEach((snapshot) => this.snapshotService.markForRemoval(snapshot).subscribe());
```
Comment (zolotyx) [inline src/app/shared/services/volume.service.ts]: This should not work like this. We should make all the async requests. and subscribe to the observable

File: src/app/shared/services/volume.service.ts [hunk 3]
```diff
@@ -132,6 +132,7 @@ export class VolumeService extends BaseBackendService<Volume> {
   }
 
   public markForRemoval(volume: Volume): Observable< Volume> {
+    volume.snapshots.forEach((snapshot) => Observable.forkJoin(this.snapshotService.markForRemoval(snapshot)).subscribe());
```
Comment (zolotyx) [inline src/app/shared/services/volume.service.ts]: Please remove subscribe

File: src/app/shared/services/volume.service.ts [hunk 4]
```diff
@@ -1,12 +1,12 @@
-import { Injectable } from '@angular/core';
-import { Observable } from 'rxjs/Observable';
-import { Subject } from 'rxjs/Subject';
-import { BackendResource } from '../decorators';
-import { Snapshot, Volume } from '../models';
-import { AsyncJobService } from './async-job.service';
-import { BaseBackendService } from './base-backend.service';
-import { SnapshotService } from './snapshot.service';
-import { VolumeTagService } from './tags/volume-tag.service';
+import {Injectable} from '@angular/core';
```
Comment (zolotyx) [inline src/app/shared/services/volume.service.ts]: Please fix code style

**Pull request merged**